src/metabase/api/card.clj

```diff
@@ -1140,34 +1158,41 @@ saved later when it is ready."
   (let [db-id             (public-settings/uploads-database-id)
         database          (or (t2/select-one Database :id db-id)
                               (throw (Exception. (tru "The uploads database does not exist."))))
+        driver            (driver.u/database->driver database)
         schema-name       (public-settings/uploads-schema-name)
+        _check-schema     (or (syncable-schema? driver database schema-name)
```

Comment:
awesome, but I think `when` is more idiomatic than `or` when we're disinterested in the return value
```suggestion
        _check-schema     (when (syncable-schema? driver database schema-name)
```

Reply:
`or` is more like `when-not` than `when`. Regardless, I think `or` is a relatively common idiom (unless I'm biased by my Ruby background?), and one point in its favor is that  it works nicely for how the code _now_ is, which has multiple clauses.

I definitely prefer

```
(or (nil? schema-name)
    (driver.s/include-schema? database schema-name)
    (throw (ex-info (tru "The schema {0} is not syncable." schema-name)
```

over

```
(when-not (or (nil? schema-name) (include-schema? ...))
  (throw ...))
  ```

Reply:
Yes, sorry, I meant `when-not`. I haven't seen the `or` idiom used in clojure before because it has `when` and `when-not`, which is generally an improvement over `or` because it signals to the reader that the returned value of the conditional expression is never used. Here you're signalling that with `_` in front of `_check-schema` so that argument doesn't apply here.

I definitely prefer 
```
(when (and schema-name (not (include-schema? ...))
  (throw ...))
```
over
```
(or (nil? schema-name)
    (driver.s/include-schema? database schema-name)
    (throw (ex-info (tru "The schema {0} is not syncable." schema-name)
```
because (1) `or` expressions are so rarely used for this purpose in clojure, and (2) this expression reads more like how you would express it in english.

```
(when (and schema-name (not (include-schema? ...))
  (throw ...))
```
Translates in my mind to: "When there is a schema-name and it's not syncable, throw an error". This sentence sounds pretty natural.

But 
```
(or (nil? schema-name)
    (driver.s/include-schema? database schema-name)
    (throw (ex-info (tru "The schema {0} is not syncable." schema-name)
```
Translates to: "Unless schema-name is nil, or schema-name is syncable, throw an error". Personally I have to do more mental gymnastics to figure out what this means.